.github/workflows/deploy.yaml

```diff
@@ -25,7 +42,8 @@ jobs:
           - name: Production
     environment: ${{ matrix.environment.name }}
     concurrency:
-      group: ${{ github.workflow }}-${{ github.job }}-${{ matrix.environment.name }}
+      group: ${{ github.workflow }}-synchronize-${{ matrix.environment.name }}-${{ needs.prepare.outputs.component-name }}
+      cancel-in-progress: true
```

Comment:
Workflow status is `waiting` when waiting for approvals and (presumably) during the wait timer so this is necessary. Otherwise new jobs will be pending until previous workflows have completed or expired. 

This may be a problem if multiple workflows are created in quick succession and cancel each other while actually running but we'll figure it out when it gets to that point.

Reply:
Does this seem like it will fix the "stuck waiting for workflows" problem I saw in #2841?

Reply:
I'm not sure why #2841 had waiting workflows. It's possible it erroneously created the deploy workflow so I've add a branch filter but there's no strong evidence of that from the actions history